src/Workspaces/Core/Portable/Editing/ImportAdderService.cs

```diff
@@ -142,17 +145,19 @@ private SyntaxNode GenerateNamespaceImportDeclaration(INamespaceSymbol namespace
         /// </summary>
         /// <param name="root">ref as we add simplifier annotations to nodes with explicit namespaces</param>
         /// <returns></returns>
-        private (ImmutableArray<SyntaxNode> imports, IEnumerable<INamespaceSymbol> namespaceSymbols, SyntaxNode? context) GetImportDirectivesFromSyntaxes(
+        private async ValueTask<(ImmutableArray<SyntaxNode> imports, IEnumerable<INamespaceSymbol> namespaceSymbols, SyntaxNode? context, SemanticModel model, SyntaxNode? newRoot)> GetImportDirectivesFromSyntaxesAsync(
```

Comment:
it deeply bothers me that this returns an IEnumerable.

Reply:
I was only changing the `SemanticModel` portion here. The enumerable could be addressed separately, but I did not see a reason to do that from a performance perspective.

Reply:
it becomes necessary for me because it crosses an async boundary.  and now i can't tell who actually performs the computation for it for me.  we've strongly held (and been bittn by) that we shouldn't have async+IEnumerable.

Reply:
Makes sense. I clarified the return value as eagerly computed.